app/Jobs/GenerateSimilarityIndex.php

```diff
@@ -202,26 +185,15 @@ protected function getOutputJsonPath(Volume $volume)
      * @param array $imagesHashArray
      * @return string Path to the JSON file.
      */
-    protected function createInputJson(Volume $volume, Array $imagesHashArray)
+    protected function createInputJson(Volume $volume, $imagesHashArray)
     {
         $path = $this->getInputJsonPath($volume);
-        $content = json_encode([$imagesHashArray]);
+        $content = json_encode($imagesHashArray);
```

Comment:
As this is a Laravel collection, you could also do:
```suggestion
        $content = $imagesHashArray->toJson();
```